### PR TITLE
Pickles GUI: Generate equivalent PowerShell or Windows Console command line

### LINKED ISSUE #90: Could the GUI generate the equivalent PowerShell or Windows Console command line?
All my SpecFlow tests are run nightly by TeamCity. It would be really helpful for the GUI to have an automation-friendly mode which would generate a PS1 command line which could then be pasted into a TeamCity Step.

The workflow would be something like
- [ ] use GUI to tune and check Pickles settings
- [ ] use GUI to generate PowerShell (or Windows Console) command line
- [ ] paste command line into appropriate TeamCity Step
- [ ] run overnight, every night, for improved documentation
- [ ] bundle Pickles output into deployment along with release notes

### PR DESCRIPTION
**Pull request by magicmonty**
This PR will fix #90

It adds two buttons, which generate the respective commands, put them into the clipboard and show a message box with the generated commands.
For each document output type, a single command line will be generated

### CODE REVIEW

File: src/Pickles/Pickles.UserInterface.UnitTests/CommandGeneration/WhenGeneratingCLICommands.cs [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+using NFluent;
+
+using NUnit.Framework;
+
+using PicklesDoc.Pickles;
+using PicklesDoc.Pickles.UserInterface.CommandGeneration;
+using PicklesDoc.Pickles.UserInterface.Settings;
+
+namespace System.CommandGeneration
```
Comment (dirkrombauts) [inline src/Pickles/Pickles.UserInterface.UnitTests/CommandGeneration/WhenGeneratingCLICommands.cs]: The namespace doesn't fit with the rest of the project

File: src/Pickles/Pickles.UserInterface.UnitTests/CommandGeneration/WhenGeneratingPowerShellCommands.cs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+using NFluent;
+
+using NUnit.Framework;
+
+using PicklesDoc.Pickles;
+using PicklesDoc.Pickles.UserInterface.CommandGeneration;
+using PicklesDoc.Pickles.UserInterface.Settings;
+
+namespace System.CommandGeneration
```
Comment (dirkrombauts) [inline src/Pickles/Pickles.UserInterface.UnitTests/CommandGeneration/WhenGeneratingPowerShellCommands.cs]: The namespace doesn't fit with the rest of the project

File: src/Pickles/Pickles.UserInterface/ViewModel/MainViewModel.cs [hunk 3]
```diff
@@ -485,6 +505,44 @@ private bool CanGenerate()
                    && this.isLanguageValid;
         }
 
+        private void DoGenerateCommandLine(CommandGeneratorBase generator)
+        {
+            var model = this.CreateMainModel();
+            var language = this.selectedLanguage.TwoLetterISOLanguageName;
+            var commands = generator.Generate(model, language);
+            Clipboard.SetText(commands);
+            MessageBox.Show("Copied the following commands into the clipboard:\n\n" + commands, "Info", MessageBoxButton.OK, MessageBoxImage.Information);
```
Comment (dirkrombauts) [inline src/Pickles/Pickles.UserInterface/ViewModel/MainViewModel.cs]: This I don't like at all - it breaks the decoupling between the viewmodel and the view.
I would like to remove it, but then the user will have no feedback that anything happened when they click on one of the command-generating buttons. So what I suggest is that you include a logger in the class (`private static readonly Logger Log = LogManager.GetLogger(MethodBase.GetCurrentMethod().DeclaringType.Name);`) and write the text into the logger. Then the user will see the result of their action in the logging grid.

Comment (magicmonty) [reply src/Pickles/Pickles.UserInterface/ViewModel/MainViewModel.cs]: Agreed and fixed

File: src/Pickles/Pickles.UserInterface.UnitTests/CommandGeneration/WhenGeneratingPowerShellCommands.cs (hunk 2 above)
Comment (magicmonty) [reply src/Pickles/Pickles.UserInterface.UnitTests/CommandGeneration/WhenGeneratingPowerShellCommands.cs]: Fixed

File: src/Pickles/Pickles.UserInterface.UnitTests/CommandGeneration/WhenGeneratingCLICommands.cs (hunk 1 above)
Comment (magicmonty) [reply src/Pickles/Pickles.UserInterface.UnitTests/CommandGeneration/WhenGeneratingCLICommands.cs]: Fixed

**Comment by magicmonty** [comment]
@dirkrombauts Should I rebase and squash too?

**Comment by dirkrombauts** [comment]
Rebasing would be cool. No need to squash.

**Comment by magicmonty** [comment]
Ok, rebased

**Comment by dirkrombauts** [comment]
Thank you for your contribution!

**Comment by dirkrombauts** [comment]
Released in version [2.20.0](https://github.com/picklesdoc/pickles/releases/tag/v2.20.0).

**Pull request merged**